Add support for Gazebo Classic simulation

2b-t opened the pull request:
This pull request adds support for **simulation of the 3D lidar with [Gazebo Classic](https://classic.gazebosim.org/)**. The syntax for sensor plug-ins in [Ignition Gazebo](https://gazebosim.org/docs/latest/sensors/) is slightly different and I will have to adapt the URDF to work with it. There seems to be no obvious way of supporting both.
You can test this pull request by running:

```
$ rosdep install --from-paths src --ignore-src -r
$ ros2 launch magnet_description visualize.launch.py
```
and finally visualizing the lidar topic `/hesai/pandar`:

![Gazebo Classic](https://github.com/user-attachments/assets/a22fc73e-161c-4140-bbb0-0c85c4dc8355)

mauricefallon commented:
@2b-t 
!!! you cannot add a dependency on gazebo to run a simple tool to visualize a urdf !!!

most of us dont want to run gazebo but we want to test or inspect the URDF in Rviz
You are asking me to install 100s of mb and several package.

I dont have the time to babysit this kind of stuff.

mauricefallon commented:
* Make one visualizer for just viewing the urdf and another for running gazebo additionally.
* write instructions for installing the minimal set of packages to visualise the URDF.

2b-t commented:
Fair enough! I reverted the changes and opened new pull requests https://github.com/ori-drs/hesai_description/pull/7 as well as https://github.com/ori-drs/magnet_description/pull/6 addressing your comments.

Clearly the problem with Gazebo Classic as a dependency that you mentioned is only a problem when running `rosdep`. The package compiled just fine without Gazebo being present and similarly visualizing the URDF without Gazebo was always possible by either setting the `simulation` argument of `visualize.launch.py` to `false` or running `description.launch.py` and RViz manually.
